wordpress-mobile/WordPress-iOS
# Stats Insights: Optimize table view updates by integrating diffable data source

Use a diffable data source within Stats Insights and Stats Insights Details to make table view cell reloads more efficient. More information about implementation in: https://github.com/wordpress-mobile/WordPress-iOS/pull/22542

Related diff: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/751
 
## To test:

1. Open Stats Insights
2. Confirm data loads correctly without flickering
3. Tap on Insights management button on top right
4. Remove all cards
5. Confirm empty view with a button is shown
6. Add all cards
7. Tap "View More" on all the cards
8. Confirm Stats Insights Details view opens

## Regression Notes
1. Potential unintended areas of impact

Breaking loading of data within existing Stats Insights and Stats Insights Details. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)

- Manual testing
- `SiteStatsInsightsViewModelTests` should still succeed

9. What automated tests I added (or what prevented me from doing so)

PR submission checklist:


Testing checklist:
